From 565d557603231468ab71c74df383da42dc72b334 Mon Sep 17 00:00:00 2001
From: Masami Hiramatsu <mhiramat@kernel.org>
Date: Tue, 23 May 2017 15:09:01 +0900
Subject: [PATCH 16/41] selftests/ftrace: Use top-level
 available_filter_function

Use top-level available_filter_function if the test case
is running under an instance.

Signed-off-by: Masami Hiramatsu <mhiramat@kernel.org>
Signed-off-by: Shuah Khan <shuahkh@osg.samsung.com>
---
 .../selftests/ftrace/test.d/ftrace/func_traceonoff_triggers.tc        | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/testing/selftests/ftrace/test.d/ftrace/func_traceonoff_triggers.tc b/tools/testing/selftests/ftrace/test.d/ftrace/func_traceonoff_triggers.tc
index c8e02ec01eaf..1b0817bfa286 100644
--- a/tools/testing/selftests/ftrace/test.d/ftrace/func_traceonoff_triggers.tc
+++ b/tools/testing/selftests/ftrace/test.d/ftrace/func_traceonoff_triggers.tc
@@ -63,6 +63,10 @@ fi
 
 # powerpc uses .schedule
 func="schedule"
+available_file=available_filter_functions
+if [ -d ../../instances -a -f ../../available_filter_functions ]; then
+   available_file=../../available_filter_functions
+fi
 x=`grep '^\.schedule$' available_filter_functions | wc -l`
 if [ "$x" -eq 1 ]; then
    func=".schedule"
-- 
2.13.0

